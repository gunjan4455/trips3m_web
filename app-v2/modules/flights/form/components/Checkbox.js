import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Checkbox extends Component {

  toggleCheckboxChange = (value) => {
    const { input, handleOnChange } = this.props;
    input.onChange(value);
    handleOnChange(value);
  };

  render() {
    const { label, input, disabled, defaultChecked } = this.props;

    return (
      <div className="p0 option-overflow">
        <div className="flex">
          <input
            {...input}
            type="checkbox"
            id={`filter_${input.name}_${input.value}`}
            className="checkbox-common"
            onChange={this.toggleCheckboxChange}
            disabled={disabled}
            defaultChecked={defaultChecked}
          />
          <label className='break-all' htmlFor={`filter_${input.name}_${input.value}`}>
            {label}
          </label>
        </div>
      </div>
    );
  }
}

Checkbox.propTypes = {
  label: PropTypes.string.isRequired,
  handleOnChange: PropTypes.func,
  input: PropTypes.object.isRequired,
  disabled: PropTypes.bool,
  defaultChecked: PropTypes.bool
};

Checkbox.defaultProps = {
  handleOnChange: () => {},
  disabled: false,
  defaultChecked: false
};

export default Checkbox;
