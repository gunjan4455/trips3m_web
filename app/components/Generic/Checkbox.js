import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Checkbox extends Component {

  toggleCheckboxChange = () => {
    const { handleOnChange, id } = this.props;
    handleOnChange(id);
  }
  render() {
    const isChecked = this.props.isChecked || false;
    const { title, value, name } = this.props;
    return (
      <div className="col-xs-6 p0 option-overflow">
        <div className="flex">
        <input
          type="checkbox"
          name={name}
          value={value}
          id={`filter_${name}_${value}`}
          className="checkbox-common"
          checked={isChecked}
          onChange={this.toggleCheckboxChange}
          disabled={this.props.disabled}
        />
        <label className='break-all' htmlFor={`filter_${name}_${value}`}>
          {title}
        </label>
          {
            this.props.count !== undefined ?
              <span className='at_count hide iblock pl0 p8' >[{this.props.count}]</span> : null
          }
        </div>
      </div>
    );
  }
}

Checkbox.propTypes = {
  title: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  isChecked: PropTypes.bool.isRequired,
  handleOnChange: PropTypes.func.isRequired,
  id: PropTypes.number.isRequired,
  value: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  count: PropTypes.number
};

Checkbox.defaultProps = {
  disabled: false
};

export default Checkbox;
