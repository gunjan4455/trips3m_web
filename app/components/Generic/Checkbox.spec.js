import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Checkbox from './Checkbox';

configure({ adapter: new Adapter() });

const MockHandleOnChange = jest.fn();

const DEFAULT_PROPS = {
  id:2,
  isChecked:false,
  name:"hotel_ratings[]",
  title:"3 Star",
  value:"3",
  disabled:false,
  filterCountKey:"hotel_ratings_3",
  handleOnChange: MockHandleOnChange
};

const shallowWrapperCheckbox = (props) => {
  return shallow(<Checkbox {...props} />);
};

describe('Checkbox module', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperCheckbox(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should handleOnChange call properly', () => {
    const wrapper = shallowWrapperCheckbox(props);
    wrapper.instance().toggleCheckboxChange();
    expect(MockHandleOnChange).toHaveBeenCalledWith(2);
  });

  it('should not render packages count when count in props is not defined', () => {
    const wrapper = shallowWrapperCheckbox(props);  
    expect(wrapper.find('.at_count')).toHaveLength(0);
  });

  it('should render packages count when count in props is defined', () => {
    props.count = 101;
    const wrapper = shallowWrapperCheckbox(props);  
    expect(wrapper.find('.at_count')).toHaveLength(1);
  });
});
