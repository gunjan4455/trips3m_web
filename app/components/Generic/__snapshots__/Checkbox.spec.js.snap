// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    disabled={false}
    filterCountKey="hotel_ratings_3"
    handleOnChange={[MockFunction]}
    id={2}
    isChecked={false}
    name="hotel_ratings[]"
    title="3 Star"
    value="3"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="flex"
      >
        <input
          checked={false}
          className="checkbox-common"
          disabled={false}
          id="filter_hotel_ratings[]_3"
          name="hotel_ratings[]"
          onChange={[Function]}
          type="checkbox"
          value="3"
        />
        <label
          className="break-all"
          htmlFor="filter_hotel_ratings[]_3"
        >
          3 Star
        </label>
      </div>,
      "className": "col-xs-6 p0 option-overflow",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={false}
            className="checkbox-common"
            disabled={false}
            id="filter_hotel_ratings[]_3"
            name="hotel_ratings[]"
            onChange={[Function]}
            type="checkbox"
            value="3"
          />,
          <label
            className="break-all"
            htmlFor="filter_hotel_ratings[]_3"
          >
            3 Star
          </label>,
          null,
        ],
        "className": "flex",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": false,
            "className": "checkbox-common",
            "disabled": false,
            "id": "filter_hotel_ratings[]_3",
            "name": "hotel_ratings[]",
            "onChange": [Function],
            "type": "checkbox",
            "value": "3",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "3 Star",
            "className": "break-all",
            "htmlFor": "filter_hotel_ratings[]_3",
          },
          "ref": null,
          "rendered": "3 Star",
          "type": "label",
        },
        null,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="flex"
        >
          <input
            checked={false}
            className="checkbox-common"
            disabled={false}
            id="filter_hotel_ratings[]_3"
            name="hotel_ratings[]"
            onChange={[Function]}
            type="checkbox"
            value="3"
          />
          <label
            className="break-all"
            htmlFor="filter_hotel_ratings[]_3"
          >
            3 Star
          </label>
        </div>,
        "className": "col-xs-6 p0 option-overflow",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              checked={false}
              className="checkbox-common"
              disabled={false}
              id="filter_hotel_ratings[]_3"
              name="hotel_ratings[]"
              onChange={[Function]}
              type="checkbox"
              value="3"
            />,
            <label
              className="break-all"
              htmlFor="filter_hotel_ratings[]_3"
            >
              3 Star
            </label>,
            null,
          ],
          "className": "flex",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "className": "checkbox-common",
              "disabled": false,
              "id": "filter_hotel_ratings[]_3",
              "name": "hotel_ratings[]",
              "onChange": [Function],
              "type": "checkbox",
              "value": "3",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "3 Star",
              "className": "break-all",
              "htmlFor": "filter_hotel_ratings[]_3",
            },
            "ref": null,
            "rendered": "3 Star",
            "type": "label",
          },
          null,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
